### PR TITLE
Capture AzDO description of badrequest messages

### PR DESCRIPTION
**Pull request by JohnTortugo**
Relates to: https://github.com/dotnet/arcade/issues/2334

Capture AzDO description for HTTP BadRequest messages. This will help a ton debugging the issue pointed above.

### CODE REVIEW

File: TestPipelineRunner/Program.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+using Microsoft.DotNet.DarcLib;
```
Comment (JohnTortugo) [inline TestPipelineRunner/Program.cs]: Added by mistake. I'll remove this.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs [hunk 2]
```diff
@@ -65,6 +65,11 @@ public async Task<HttpResponseMessage> ExecuteAsync(int retryCount = 3)
 
                             retriesRemaining = 0;
                         }
+                        else if (response.StatusCode == HttpStatusCode.BadRequest)
+                        {
+                            var errorDetails = await response.Content.ReadAsStringAsync();
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs]: Add ` if (_logFailure)`

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs [hunk 3]
```diff
@@ -65,6 +65,11 @@ public async Task<HttpResponseMessage> ExecuteAsync(int retryCount = 3)
 
                             retriesRemaining = 0;
                         }
+                        else if (response.StatusCode == HttpStatusCode.BadRequest)
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs]: Is the plan to retry if this happens? Are we calling with 0 retries? If not, should we set `retriesRemaining` to 0?

Comment (JohnTortugo) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs]: I don't think so. That error means something is wrong in the request body. IMO retrying won't do anything.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs (hunk 2 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs]: done

**Pull request merged**